### DISCUSSION TITLE
mirai和粘液科技SlimeFun插件同时装载导致粘液科技无法使用

### DISCUSSION THREAD

**Discussion by TATyKeFei** [DreamVoid/MiraiMC#441] [问答 / Q&A]
触发条件：粘液科技与MiraiMc插件同时使用，使用粘液科技指南书尝试打开Gui界面
服务端核心：purpur1.20.1
粘液科技版本：Slimefun v2023.07-release
MiraiMc版本：1.8

控制台报错代码：

> [04:22:35 ERROR]: Error occurred while enabling Slimefun v2023.07-release (Is it up to date?)
> java.lang.NoClassDefFoundError: io/github/thebusybiscuit/slimefun4/implementation/listeners/CoolerListener
>         at io.github.thebusybiscuit.slimefun4.implementation.Slimefun.registerListeners(Slimefun.java:665) ~[[SF][粘 液科技]Slimefun-2023.07-release.jar:?]
>         at io.github.thebusybiscuit.slimefun4.implementation.Slimefun.onPluginStart(Slimefun.java:357) ~[[SF][粘液科 技]Slimefun-2023.07-release.jar:?]
>         at io.github.thebusybiscuit.slimefun4.implementation.Slimefun.onEnable(Slimefun.java:250) ~[[SF][粘液科技]Slimefun-2023.07-release.jar:?]
>         at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:281) ~[purpur-api-1.20.1-R0.1-SNAPSHOT.jar:?]
>         at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:189) ~[purpur-1.20.1.jar:git-Purpur-2026]
>         at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:104) ~[purpur-1.20.1.jar:git-Purpur-2026]
>         at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:507) ~[purpur-api-1.20.1-R0.1-SNAPSHOT.jar:?]
>         at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugin(CraftServer.java:654) ~[purpur-1.20.1.jar:git-Purpur-2026]
>         at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugins(CraftServer.java:565) ~[purpur-1.20.1.jar:git-Purpur-2026]
>         at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:641) ~[purpur-1.20.1.jar:git-Purpur-2026]
>         at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:440) ~[purpur-1.20.1.jar:git-Purpur-2026]        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:345) ~[purpur-1.20.1.jar:git-Purpur-2026]
>         at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1117) ~[purpur-1.20.1.jar:git-Purpur-2026]
>         at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:323) ~[purpur-1.20.1.jar:git-Purpur-2026]
>         at java.lang.Thread.run(Unknown Source) ~[?:?]
> Caused by: java.lang.ClassNotFoundException: io.github.thebusybiscuit.slimefun4.implementation.listeners.CoolerListener
>         at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:185) ~[purpur-api-1.20.1-R0.1-SNAPSHOT.jar:?]
>         at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:152) ~[purpur-api-1.20.1-R0.1-SNAPSHOT.jar:?]
>         at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
>         ... 15 more

求求各位大佬帮忙看看QWQ
Mirai和粘液科技这两个插件对我来说真的很重要QAQ

**Comment by TATyKeFei** [comment]
还有，mirai没有看到如何报错。所有插件都检查了一遍是装上Mirai插件后才导致粘液科技无法使用的

**Comment by cssxsh** [comment]
@DreamVoid 你是不是改了 ContextClassLoader 没改回来

**Reply by DreamVoid** [reply]
所以，issue里强制要求提供完整服务端日志是有原因的。

**Comment by DreamVoid** [comment]
# ？
![image.png](https://s2.loli.net/2023/08/09/TkjM18oDNBehyKH.png)
![image.png](https://s2.loli.net/2023/08/09/93q7lQXecpHYiLK.png)

我不知道你用的什么杂牌slimefun，我的评价是可以不用。
你应该想想为什么只有粘液科技出了问题，而其他插件什么事都没有。

**Reply by TATyKeFei** [reply]
我用的第三方的汉化版https://slimefun-wiki.guizhanss.cn/Installing-Slimefun，看下面的评论是汉化版禁止启动了，不过现在换回英文原版解决了。感谢

**Comment by ApliNi** [comment]
有一个汉化版的粘液科技，里面写了检查到 MiraiMC 插件就不启动的代码，删掉这段后粘液科技能正常运行。

**Reply by TATyKeFei** [reply]
哦原来如此，我换回英文原版好了。感谢

**Comment by cssxsh** [comment]
https://github.com/StarWishsama/Slimefun4/blame/bf49fd87bae780ca3ba56644f4c3bb51d3ccc14e/src/main/java/city/norain/slimefun4/EnvironmentChecker.java#L11-L18
